log4j-api/src/main/java/org/apache/logging/log4j/message/ParameterFormatter.java

```diff
@@ -240,14 +240,6 @@ static void formatMessage(
             return;
         }
 
-        // Fail if there are insufficient arguments
-        if (analysis.placeholderCount > args.length) {
-            final String message = String.format(
-                    "found %d argument placeholders, but provided %d for pattern `%s`",
-                    analysis.placeholderCount, args.length, pattern);
-            throw new IllegalArgumentException(message);
-        }
```

Comment:
I would prefer to issue a `StatusLogger` warning whenever the number of arguments:

- is less than `analysis.placeholderCount` (some arguments are missing),
- is higher than `analysis.placeholderCount + (args[args.length - 1] instance of Throwable ? 1 : 0)` (some arguments will be ignored).

Since there will be no more exceptions throws, you can remove the `try...catch`es at the call sites.

Reply:
I got your point, I'm modifying the codes.

Reply:
Agreed with @ppkarwasz's remark. Please provide sufficient test coverage for all possibilities he shared. In particular, test not just the unhappy flow, but also happy flow.